grassroots-backend/src/users/Users.service.ts

```diff
@@ -2,28 +2,98 @@ import { Injectable } from "@nestjs/common";
 import { UserEntity } from "./User.entity";
 import { EntityManager, EntityRepository } from "@mikro-orm/core";
 import { UserDTO } from "../grassroots-shared/User.dto";
+import { instanceToPlain } from "class-transformer";
+import { OrganizationEntity } from "../organizations/Organization.entity";
+import { Permission } from "../grassroots-shared/Permission";
+import { OrganizationRepository } from "../organizations/Organization.repo";
+import { UserRoleEntity } from "./UserRole.entity";
 
 @Injectable()
 export class UsersService {
   repo: EntityRepository<UserEntity>;
+  organizationRepo: OrganizationRepository;
   constructor(private readonly entityManager: EntityManager) {
     this.repo = entityManager.getRepository<UserEntity>(UserEntity);
+    this.organizationRepo =
+      entityManager.getRepository<OrganizationEntity>(OrganizationEntity);
   }
-  async findOrCreate(user: UserDTO): Promise<UserDTO | undefined> {
+  async findOrCreate(user: UserDTO): Promise<UserDTO> {
     const existing = await this.repo.findOne({ id: user.id });
     if (existing !== null) {
-      return existing;
+      return existing.toDTO();
     }
-    const result = this.repo.create(user);
+    const newUser = this.repo.create({
+      ...instanceToPlain(user),
+      userRoles: [],
+    });
+
+    const userRoleRepo =
+      this.entityManager.getRepository<UserRoleEntity>(UserRoleEntity);
```

Comment:
Looking over this again, I think it probably makes sense to store a reference to this.

Reply:
Done.